### DISCUSSION TITLE
useWaitForTransaction always response success

### DISCUSSION THREAD

**Discussion by jcflorville** [wevm/wagmi#943] [Q&A] [answered]
Hi! I don't know if a issue but `useWaitForTransaction` is returning success always 

```
const { write: enterBid, data: bidData } = useContractWrite({ ...contractInfo }, ENTER_BID_COLLECTIBLE);
 
const { data: enterBidTransaction } = useWaitForTransaction({
    hash: bidData?.hash,
    onSuccess(data) {
      console.log('Succes', data);
    },
    onError(error) {
      console.log('Error', error);
    },
  });
```

i tried with `wait`

```
const { data: enterBidTransaction } = useWaitForTransaction({
    wait: bidData?.wait,
    onSuccess(data) {
      console.log('Succes', data);
    },
    onError(error) {
      console.log('Error', error);
    },
  });
```

but the same response `success`

Transaction example, this failed but the return logs return: https://goerli.etherscan.io/tx/0x6dcdbf4af7b2775ad7eee82b2ea3f10e1c5a3316f6b48b9e04aa2d88d6f23b47

Any thoughts?

**Comment by SpasiboKojima** [comment]
I have the same issue, I'd put it into Issue not Discussions.
Probably out of scope but mobile-metamask was also showing transactions as successful on both prepare screen and after they were submitted even when an error was thrown from solidity require(). But onSuccess was firing for me even on Desktop with proper failed TX acknowledgment from MM.

**Comment by tmm** [comment, marked as answer]
`useWaitForTransaction` `onSuccess` and `onError` reflect the HTTP network state (e.g. HTTP request failed) versus EVM network state (e.g. transaction failed to mine). Can look into updating this, but you should inspect the `data` from `onSuccess` to find out if the transaction failed. `data` is an ethers [`TransactionReceipt`](https://docs.ethers.io/v5/api/providers/types/#providers-TransactionReceipt) with `status=1` if successful or `status=0` if it was reverted.

**Reply by 0xCardiE** [reply]
Would be really nice to have a revert reason when transaction fails, but if I understand correctly this is bounded by TransactionReceipt  from ethers which doesn't provide it, so I guess all we can do is another call to chain and get details then like here?
https://github.com/wagmi-dev/wagmi/discussions/317
